src/test/query-users.test.ts

```diff
@@ -0,0 +1,269 @@
+  });
+
+  it('should return the default values if skip and limit are null', async () => {
+    const variables = {
+      input: {
+        skip: 0,
+        limit: null,
+      },
+    };
+
+    const response = await axios.post(
+      'http://localhost:4000/',
+      { query, variables },
+      { headers: { Authorization: token } },
+    );
+
+    const userData = response.data.data.users;
+
+    expect(userData.count).to.be.eq(count);
+    expect(userData.hasBefore).to.be.false;
+    expect(userData.hasAfter).to.be.true;
+
+    expect(userData.users[0].id).to.be.eq(user[0].id);
+    expect(userData.users[0].name).to.be.eq(user[0].name);
+    expect(userData.users[0].email).to.be.eq(user[0].email);
+    expect(userData.users[0].birthDate).to.be.eq(user[0].birthDate);
+  });
+
+  it('should return all users if limit is greater than the total number of users', async () => {
+    const variables = {
+      input: {
+        skip: 0,
+        limit: count + 5,
+      },
+    };
+
+    const response = await axios.post(
+      'http://localhost:4000/',
+      { query, variables },
+      { headers: { Authorization: token } },
+    );
+
+    const userData = response.data.data.users;
+
+    expect(userData.count).to.be.eq(count);
+    expect(userData.hasBefore).to.be.false;
+    expect(userData.hasAfter).to.be.false;
+
+    expect(userData.users[0].id).to.be.eq(user[0].id);
+    expect(userData.users[0].name).to.be.eq(user[0].name);
+    expect(userData.users[0].email).to.be.eq(user[0].email);
+    expect(userData.users[0].birthDate).to.be.eq(user[0].birthDate);
+  });
+
+  it('should return [] if skip is greater than the total number of users', async () => {
+    const variables = {
+      input: {
+        skip: count + 5,
+        limit: validLimit,
+      },
+    };
+
+    const response = await axios.post(
+      'http://localhost:4000/',
+      { query, variables },
+      { headers: { Authorization: token } },
+    );
+
+    const userData = response.data.data.users;
+
+    expect(userData.count).to.be.eq(count);
+    expect(userData.hasBefore).to.be.true;
+    expect(userData.hasAfter).to.be.false;
+    expect(userData.users).to.be.deep.eq([]);
+  });
+
+  it('should throw error if limit is 0', async () => {
+    const variables = {
+      input: {
+        skip: validSkip,
+        limit: 0,
+      },
+    };
+
+    const response = await axios.post(
+      'http://localhost:4000/',
+      { query, variables },
+      { headers: { Authorization: token } },
+    );
+
+    expect(response.data.errors).to.have.lengthOf(1);
+    const error = response.data.errors[0];
+    expect(error.message).to.be.eq('Limit não pode ser zero');
+    expect(error.code).to.be.eq(400);
+    expect(response.data.data).to.be.eq(null);
+  });
+
+  it('should throw error if user is not authenticated', async () => {
+    const variables = {
+      input: {
+        skip: validSkip,
+        limit: validLimit,
+      },
+    };
```

Comment:
como limit e skip nao sao obrigatorios e esse it está testando um fluxo antes dos inputs, nesses casos vc pode faclitar a sua vida e nem injetar os dados das variables

```suggestion
    const variables = { input: {} };
```

Reply:
fora que deixa claro que o teste nao depende de um input para funcionar